### PR TITLE
fix nix-build invocation in non-flake builds

### PR DESCRIPTION
**Pull request by stuebinm**
The `drvpath^out` syntax is only part of `nix build`, not `nix-build`. The latter still behaves as before Nix 2.15 and produces the outPath, but produces an error if one attempts to call it the same way as `nix build`.

### CODE REVIEW

**Comment by PhilTaken** [comment]
I wonder if we could set up some nixos tests to make sure there are no regressions. Would you mind looking into it?

**Comment by stuebinm** [comment]
@PhilTaken I've attempted to write a test for this (changing the existing test setup slightly to allow for non-flake-enabled nix inside the vms & their configuration, and adding a flake-compat shim).

However, it seems to hang on the `deploy` step for me for an unreasonable amount of time. I'm unsure if this is an actual issue in the test — the existing tests using flakes likewise take very long, although they do eventually succeed.

I might have another look at this later, when I have access to more powerful hardware than my laptop.

**Comment by stuebinm** [comment]
ah i forgot about doing this back then. so here we go:

the problem with the test seems to be that nix uses a surprising amount of memory to evaluate anything inside the vm (and more so without flakes); with the limit given in `nix/tests/common.nix` it becomes very slow and eventually either runs out of memory or into the default timeout of `client.succeed()` and then aborts. With an increased memory limit for the client vm, running the test takes a reasonable time (and the test succeeds).

I've also been using this patch locally for over a month now to deploy a couple of my own machines.

**Comment by rvem** [comment]
> With an increased memory limit for the client vm, running the test takes a reasonable time (and the test succeeds).

Sounds like a good enough way to resolve the issue:)

**Review by rvem — APPROVED** [review]

File: nix/tests/default.nix [hunk 1]
```diff
@@ -35,8 +35,9 @@ let
         ];
       };
       client = { nodes, ... }: {
-        imports = [ (import ./common.nix { inherit inputs pkgs; }) ];
+        imports = [ (import ./common.nix { inherit inputs pkgs flakes; }) ];
         environment.systemPackages = [ pkgs.deploy-rs.deploy-rs ];
+        virtualisation.memorySize = lib.mkForce 4096;
```
[the remark this answers is not in the document]
Comment (stuebinm) [reply nix/tests/default.nix:41]: done

**Pull request merged**